src/harness/platforms/sdl12.c

```diff
@@ -0,0 +1,236 @@
+#include <SDL.h>
+
+#include "harness.h"
+#include "harness/config.h"
+#include "harness/hooks.h"
+#include "harness/trace.h"
+#include "sdl12_scancode_to_dinput.h"
+
+#ifdef _WIN32
+#include <windows.h>
+#endif
+SDL_Surface* screen;
+//SDL_Renderer* renderer;
+//SDL_Texture* screen_texture;
+uint32_t converted_palette[256];
+br_pixelmap* last_screen_src;
+int render_width, render_height;
+
+Uint32 last_frame_time;
+
+uint8_t directinput_key_state[SDLK_LAST];
+
+static void* create_window_and_renderer(char* title, int x, int y, int width, int height) {
+    int video_flags = 0;
+    render_width = width;
+    render_height = height;
+
+    initializeSDL12KeyNums();
+
+    if (SDL_Init(SDL_INIT_VIDEO) != 0) {
+        LOG_PANIC("SDL_INIT_VIDEO error: %s", SDL_GetError());
+    }
+
+    if (harness_game_config.start_full_screen) {
+        video_flags |= SDL_FULLSCREEN;
+    }
+
+    screen = SDL_SetVideoMode(width, height, 32, video_flags);
+    SDL_WM_SetCaption("Carmageddon", NULL);
+
+    if (screen == NULL) {
+        LOG_PANIC("SDL_SetVideoMode failed (%s)", SDL_GetError());
+    }
+
+//    renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_PRESENTVSYNC);
+//    if (renderer == NULL) {
+//        LOG_PANIC("Failed to create renderer: %s", SDL_GetError());
+//    }
+//    SDL_SetRenderDrawBlendMode(renderer, SDL_BLENDMODE_NONE);
+//    SDL_RenderSetLogicalSize(renderer, render_width, render_height);
+
+//    screen_texture = SDL_CreateTexture(renderer, SDL_PIXELFORMAT_ARGB8888, SDL_TEXTUREACCESS_STREAMING, width, height);
+//    if (screen_texture == NULL) {
+//        SDL_RendererInfo info;
+//        SDL_GetRendererInfo(renderer, &info);
+//        for (Uint32 i = 0; i < info.num_texture_formats; i++) {
+//            LOG_INFO("%s\n", SDL_GetPixelFormatName(info.texture_formats[i]));
+//        }
+//        LOG_PANIC("Failed to create screen_texture: %s", SDL_GetError());
+//    }
+
+    return screen;
+}
+
+static int set_window_pos(void* hWnd, int x, int y, int nWidth, int nHeight) {
+    // SDL_SetWindowPosition(hWnd, x, y);
+    if (nWidth == 320 && nHeight == 200) {
+        nWidth = 640;
+        nHeight = 400;
+    }
+    SDL_UpdateRect(hWnd, x, y, nWidth, nHeight);
```

Comment:
This is causing garbage on screen after exit on AmigaOS. Is it needed?

Reply:
Perhaps the real issue we're nowhere calling `SDL_DestrowWindow` (or `SDL_FreeSurface` in SDL1)?

I created this using sdl12-compat on Linux. I'm going to try "real" sdl1 now.

Reply:
I think this issue got introduced by ca1ef76e7f2beef062394b2b16c2d5149c668c5f:

`PDShutdownSystem` of  `pc-win95/win95sys.c` more complete, but it got disabled by this commit:
https://github.com/dethrace-labs/dethrace/blob/2e31fc6956a5a10598d6afe023af8bbedc3278ba/src/DETHRACE/pc-win95/win95sys.c#L1

Reply:
I think this is a regression introduced by the 3dfx patch, causing `DestroyWindow` not being called.
I've created a ticket at https://github.com/dethrace-labs/dethrace/issues/446

Reply:
Since this is not a regression, I'd like to address #446 in a later pr.
I've got something local that addresses #446, but I'd like this one merged first.